src/docks/timelinedock.cpp

```diff
@@ -680,23 +679,36 @@ void TimelineDock::selectMultitrack()
     emit selected(m_model.tractor());
 }
 
-void TimelineDock::copyClip(int trackIndex, int clipIndex)
+void TimelineDock::copyClip()
 {
-    if (trackIndex < 0)
-        trackIndex = currentTrack();
-    if (clipIndex < 0)
-        clipIndex = clipIndexAtPlayhead(trackIndex);
-    Q_ASSERT(trackIndex >= 0 && clipIndex >= 0);
-    QScopedPointer<Mlt::ClipInfo> info(getClipInfo(trackIndex, clipIndex));
-    if (info) {
-        QString xml = MLT.XML(info->producer);
-        Mlt::Producer p(MLT.profile(), "xml-string", xml.toUtf8().constData());
-        p.set_speed(0);
-        p.seek(info->frame_in);
-        p.set_in_and_out(info->frame_in, info->frame_out);
-        MLT.setSavedProducer(&p);
-        emit clipCopied();
+    if (selection().isEmpty())
+            return;
+
+    setCopiedFromPlaylist(false);
+
+    m_producers.clear();
+
+    //preserve the order of the clips in the timeline
+    QList<QPoint> selected = selection();
+    std::sort(selected.begin(), selected.end(),
+              [] (const QPoint& clip1, const QPoint& clip2){return clip1.x() > clip2.x();});
+
+    for (const auto& clip : selected) {
+        QScopedPointer<Mlt::ClipInfo> info(getClipInfo(clip.y(), clip.x()));
+        if (info) {
+            QString xml = MLT.XML(info->producer);
+            Mlt::Producer p(MLT.profile(), "xml-string", xml.toUtf8().constData());
+            p.set_speed(0);
+            p.seek(info->frame_in);
+            p.set_in_and_out(info->frame_in, info->frame_out);
+            if (m_producers.isEmpty()) {
+                MLT.setSavedProducer(&p);
```

Comment:
I am not sure I quite understand what you mean.
When I copy clip(s) the Source player should have the ability to show the copied clip(s)? Like the Project playlist does with the playlist.

Reply:
Yes. Look at today:
* Open a project that uses the timeline. Copy a timeline clip. Switch to the Source player: it shows what was copied. When using timeline editing operations Paste or Overwrite, it uses whatever is in Source for that operation. Source is not only populated by a copy operation. That needs to stay the same.
* Open a project that uses the playlist (can include timeline as well). Select a playlist item and press <kbd>Shift</kbd>+<kbd>Enter</kbd>. The Project player is now showing the playlist with the playhead at the beginning of the playlist item: it has numeric markers. This currently works this way for a playlist, but a single track could also be shown this way. This is not absolutely required. If the selected clips span more than one track, then it cannot really be shown in this manner. However, it would not be in the Project player but rather the Source player would still need to show the selected subset. It is just that a subset all from a single track could show these numeric markers.

I need to feel confident that someone altering or extending existing Shotcut features really understands Shotcut.

Reply:
> I need to feel confident that someone altering or extending existing Shotcut features really understands Shotcut.

About that, a while ago when you requested me to try to fix some bugs, I went and looked at some bugs in issues and in the forum and realized that to be able to keep working on Shotcut or fix bugs I have to use it more so I can fully understand how it works by using as many of its features as possible. To this day I have used Shotcut for only one project (one which is too simple), I do have some projects that I want to start and use Shotcut but I have been really busy lately with a lot of school assignments and exams so I haven't had much time left.
If you hadn't reviewed this PR it would still be a while before I got back to work on Shotcut. I still have a lot of on plate but since I am the one who made this PR I want to address every issue you point out asap.

On a side note, referring [to this branch](https://github.com/hbtalha/shotcut/tree/createAdditionalPlaylists) and [this demostration video](https://youtu.be/RBCk1KYU4TI) I have added the feature to add multiple playlists and I got stuck when I went to try to know how to save it to the file and read it back too so I put that branch aside and worked on supportMultiSelectionInCutCopyPaste. 
Also besides saving it to and reading it from the file the code still requires adjustments due to the changes made to support additional playlists. 
I would like to know if I can create a PR or post it on the forum to share it so it can serve maybe as a starting point for some other dev to provide this feature as it may be a while until I get back to work on it and because I wll need some help with it.

Reply:
> I have added the feature to add multiple playlists

Sorry but I am not interested in exactly that, and there is not currently a straightforward way to store multiple playlists in XML and retrieve them via API. I have planned to add a tag systems with a way to filter by tags to avoid the need to have multiple playlists. See [road map](https://shotcut.org/roadmap/) item 5. I prefer you not post it to the forum. At the MLT level, tags would simply be stored in a shotcut-specific property on a producer/chain.